include/layers/EWLayer.hpp

```diff
@@ -87,6 +87,11 @@ std::vector<ValueType> EWLayerImpl<ValueType>::run(
              static_cast<ValueType>(beta_);
     };
     std::transform(input.begin(), input.end(), res.begin(), linear);
+  } else if (func_ == "sigmoid") {
+    auto sigmoid = [&](const ValueType& value) -> ValueType {
```

Comment:
Use a numerically stable sigmoid (prevents overflow/underflow for large |x|)
Replace the body with:
```
auto sigmoid = [](ValueType x) -> ValueType {
  if (x >= static_cast<ValueType>(0)) {
    ValueType z = std::exp(-x);
    return static_cast<ValueType>(1) / (static_cast<ValueType>(1) + z);
  } else {
    ValueType z = std::exp(x);
    return z / (static_cast<ValueType>(1) + z);
  }
};
```